### PR TITLE
Switch to cargo workspace

### LINKED ISSUE #548: Create cargo workspace
I would like to create a cargo workspace for this project and divide up the code into several crates
- common
- container
- ffi
- image
- network
- sandbox
- server
- services
- storage

### LINKED ISSUE #549: Switch to oci-spec-rs
We should switch to the [oci-spec](https://crates.io/crates/oci-spec) crate rather than having our own implementation of the runtime spec.

### PR DESCRIPTION
**Pull request by Furisto**
#### What type of PR is this?
/kind cleanup
/kind dependency-change

#### What this PR does / why we need it:
This switches the project to a cargo workspace. In the process I also updated the project to use the oci-spec crate.

#### Which issue(s) this PR fixes:
Fixes #548
Fixes #549

#### Special notes for your reviewer:
Rust analyzer was not able to resolve the path in the include! for the cri api in the tests crate (the rust compiler could do so successfully). In the end I symlinked it, which satisfied rust analyzer.

#### Does this PR introduce a user-facing change?
```release-note
Split api surface into multiple crates
```

### CODE REVIEW

**Comment by utam0k** [comment]
I was going to do a review, I couldn't decide if owners or something else was correct for me. 
@saschagrunert So can I ask for a review on this PR?

**Comment by Furisto** [comment]
@utam0k Looks like @saschagrunert is on vacation. Can you start the the CI and do the review? If there is anything he is not ok with we can take care of it after he is back.

**Comment by utam0k** [comment]
> @utam0k Looks like @saschagrunert is on vacation. Can you start the the CI and do the review? If there is anything he is not ok with we can take care of it after he is back.

Oh, I'm sorry about that. I understand. It's a little difficult to judge, but we'll fix that later. Can you resolve the conflicts?

File: OWNERS [hunk 1]
```diff
@@ -1,8 +1,5 @@
 approvers:
-  - giuseppe
-  - haircommander
-  - harche
-  - mrunalp
```
Comment (utam0k) [inline OWNERS:5]: If we are going to keep @mrunalp as the author of `Cargo.toml`, I think we need to keep him in OWNERS as well.

Comment (Furisto) [reply OWNERS:5]: @utam0k The idea was to list only the current active maintainers in the owners file but I left him in as author as he has contributed significantly in the past and I wanted to preserve that. I did not think a lot about this though, so I am happy to do any change you suggest.

**Review by utam0k — APPROVED** [review]

**Comment by utam0k** [comment]
@Furisto Can I ask you to add your signature to your all commits?
https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/about-commit-signature-verification

**Comment by Furisto** [comment]
@utam0k Done

**Comment by utam0k** [comment]
@Furisto  This commit does not seem to be done. d236dd7ec43014df93e6d6ac7ad64e1d3cb57cf3

**Comment by Furisto** [comment]
@utam0k Done

**Comment by utam0k** [comment]
Great!

**Pull request merged**